AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/SharedPrefHandler.java

```diff
@@ -322,10 +322,21 @@ private IJExpression extractResValue(SharedPrefHolder holder, Element method, IR
 		case STRING:
 			resourceGetMethodName = "getString";
 			break;
+		case ARRAY:
+			resourceGetMethodName = "getStringArray";
+			break;
 		default:
 			break;
 		}
-		return holder.getContextField().invoke("getResources").invoke(resourceGetMethodName).arg(idRef);
+
+		JInvocation resourceInvocation = holder.getContextField().invoke("getResources").invoke(resourceGetMethodName).arg(idRef);
+
+		if (IRClass.Res.ARRAY.equals(res)) {
+			JInvocation asList = getClasses().ARRAYS.staticInvoke("asList");
+			JInvocation newHashMap = JExpr._new(getClasses().HASH_SET.narrow(getClasses().STRING));
+			resourceInvocation = newHashMap.arg(asList.arg(resourceInvocation));
```

Comment:
emtpy string array resource should not be a prolem. this code generates `new HashSet<>(Arrays.asList(res.getStringArray(R.array.xxx)));` if the resource is empty, the array is emtpy, the list is empty and the set is empty. also i think we have no way to access the number of items of a resource on compile time as the value might be diffrent for e.g. locales or screen sizes.